// gatsby-source-sanity-transform-images/gatsby-node.js
"use strict";

exports.__esModule = true;
exports.createResolvers = exports.setFieldsOnGraphQLNodeType = void 0;

var _graphql = require("gatsby/graphql");

var _gatsbySourceFilesystem = require("gatsby-source-filesystem");

var _types = require("./types");

const setFieldsOnGraphQLNodeType = ({
  type
}) => {
  if (type.name === `SanityImageAsset`) {
    return {
      localFile: {
        type: `File`,
        args: {
          width: {
            type: _graphql.GraphQLInt
          },
          format: {
            type: _types.ImageFormatType
          },
          height: {
            type: _graphql.GraphQLInt
          },
          fit: {
            type: _graphql.GraphQLString,
            defaultValue: "crop"
          }
        }
      }
    };
  } // by default return empty object


  return {};
};

exports.setFieldsOnGraphQLNodeType = setFieldsOnGraphQLNodeType;

const createResolvers = ({
  actions: {
    createNode
  },
  cache,
  createNodeId,
  createResolvers,
  store,
  reporter
}) => {
  const resolvers = {
    SanityImageAsset: {
      localFile: {
        resolve: (source, {
          width,
          height,
          fit,
          format
        }) => {
          const url = `${source.url}?fit=${fit}${format ? `&fm=${format}` : ''}${width ? `&w=${width}` : ''}${height ? `&h=${height}` : ''}`;
          return (0, _gatsbySourceFilesystem.createRemoteFileNode)({
            url,
            store,
            cache,
            createNode,
            createNodeId,
            reporter
          });
        }
      }
    }
  };
  createResolvers(resolvers);
};

exports.createResolvers = createResolvers;
